File: Java/Targeta.java
import java.util.Date;
class Efectivo extends Payment{
        String franquicia;
        Date fechaVencimiento;
        String codigoCvv;


        public Efectivo(Integer id, String franquicia, Date fechaVencimiento, String codigoCvv){
        super(id);
        this.franquicia = franquicia;
        this.fechaVencimiento = fechaVencimiento;
        this.codigoCvv = codigoCvv;
        }
}

File: PHP/Targeta.php
<?php
require_once('Payment.php');

class Targeta extends Payment{
    public $franquicia;
    public $fechaVencimiento;
    public $cvv;

    public function __construct($id, $franquicia, $fechaVencimiento, $cvv){
        parent::__construct($id);
        $this->franquicia = $franquicia;
        $this->fechaVencimiento = $fechaVencimiento;
        $this->cvv = $cvv;
    }
}
?>